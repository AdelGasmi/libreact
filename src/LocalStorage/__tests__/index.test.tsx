import {createElement as h} from 'react';
import {mount} from 'enzyme';
import {LocalStorage} from '..';

const glob = global as any;

const promisify = (assertions, delay = 20) => {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      try {
        assertions();
        resolve();
      } catch (error) {
        reject(error);
      }
    }, delay);
  });
};

describe('<LocalStorage>', () => {
  beforeEach(() => {
    glob.localStorage = {};
  });

  it('is a component', () => {
    expect(LocalStorage).toBeInstanceOf(Function);
  });

  it('puts value to localStorage', () => {
    const wrapper = mount(<LocalStorage name='foo' data='bar' debounce={10} />);

    return promisify(() => {
      expect(glob.localStorage).toMatchSnapshot();
    });
  });

  it('serializes an object', () => {
    const wrapper = mount(<LocalStorage name='foo' data={{key: [1, 2, 3]}} debounce={10} />);

    return promisify(() => {
      expect(glob.localStorage).toMatchSnapshot();
    });
  });

  it('updates on re-render', () => {
    const wrapper = mount(<LocalStorage name='foo' data={1} debounce={10} />);

    wrapper.setProps({
      name: 'foo',
      data: 2
    });

    wrapper.update();

    return promisify(() => {
      expect(glob.localStorage.foo).toBe('2');
    }, 50);
  });

  it('does NOT set initial data when onMount prop is set', () => {
    const wrapper = mount(<LocalStorage name='foo' data={1} debounce={10} onMount={() => {}} />);

    return promisify(() => {
      expect(glob.localStorage).toMatchSnapshot();
    }, 20);
  });
});
