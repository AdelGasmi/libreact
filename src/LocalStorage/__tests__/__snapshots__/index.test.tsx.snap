// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocalStorage> does NOT set initial data when onMount prop is set 1`] = `Object {}`;

exports[`<LocalStorage> puts value to localStorage 1`] = `
Object {
  "foo": "\\"bar\\"",
}
`;

exports[`<LocalStorage> serializes an object 1`] = `
Object {
  "foo": "{\\"key\\":[1,2,3]}",
}
`;
